Wikipedia/Code/ArticlePeekPreviewViewController.swift

```diff
@@ -0,0 +1,94 @@
+import UIKit
+import WMF
+
+@objc (WMFArticlePeekPreviewViewController)
+class ArticlePeekPreviewViewController: UIViewController {
+    
+    fileprivate let articleURL: URL
+    fileprivate let dataStore: MWKDataStore
+    fileprivate var theme: Theme
+
+    @IBOutlet weak var leadImageView: UIImageView!
+    @IBOutlet weak var headerView: UIView!
+    @IBOutlet weak var titleLabel: UILabel!
+    @IBOutlet weak var descriptionLabel: UILabel!
+    @IBOutlet weak var textLabel: UILabel!
+    
+    @objc required init(articleURL: URL, dataStore: MWKDataStore, theme: Theme) {
+        self.articleURL = articleURL
+        self.dataStore = dataStore
+        self.theme = theme
+        super.init(nibName: "ArticlePeekPreviewViewController", bundle: nil)
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        return nil
+    }
+    
+    func fetchArticle() {
+        guard let article = dataStore.fetchArticle(with: articleURL) else {
+            dataStore.viewContext.wmf_updateOrCreateArticleSummariesForArticles(withURLs: [articleURL], completion: { (articles) in
+                guard let first = articles.first else {
+                    return
+                }
+                self.updateView(with: first)
+            })
+            return
+        }
+        updateView(with: article)
+    }
+    
+    func updateView(with article: WMFArticle) {
+        
+        if let imageURL = article.imageURL(forWidth: traitCollection.wmf_leadImageWidth) {
+            self.leadImageView.wmf_setImage(with: imageURL, detectFaces: true, onGPU: true, failure: { (error) in
+                self.leadImageView.isHidden = true
+            }, success: {
+                //handle success
+            })
+        } else {
+            leadImageView.isHidden = true
+        }
+
+        self.titleLabel.text = article.displayTitle
+        self.descriptionLabel.text = article.capitalizedWikidataDescription
+        self.textLabel.text = article.snippet
+        
+        self.preferredContentSize = self.view.systemLayoutSizeFitting(CGSize(width: self.view.bounds.size.width, height: UILayoutFittingCompressedSize.height), withHorizontalFittingPriority: UILayoutPriority.required, verticalFittingPriority: UILayoutPriority.fittingSizeLevel)
+        self.parent?.preferredContentSize = self.preferredContentSize
+    }
+    
+    override func viewDidLoad() {
+        fetchArticle()
+        updateFonts()
```

Comment:
Does dynamic type work as expected? `updateFonts()` might need to be in `viewWillAppear` so that the trait collection is up-to-date. If it's working fine, don't bother moving it.